File: 3540-hash-divided-string/hash-divided-string.py
class Solution:
    def stringHash(self, s: str, k: int) -> str:
        n = len(s)
        alpha = 'abcdefghijklmnopqrstuvwxyz'
        ans = ''

        for i in range(0, n, k):
            substr = s[i : i + k]
            hsum = 0

            for char in substr:
                hsum += alpha.index(char)

            hsum %= 26

            ans += alpha[hsum]

        return ans

File: 3540-hash-divided-string/hash-divided-string.cpp
class Solution {
public:
    string stringHash(string s, int k) {
        string alpha = "abcdefghijklmnopqrstuvwxyz";
        string ans = "";
        int n = s.length();

        for (int i = 0; i < n; i += k) {
            int hsum = 0;
            string substring = s.substr(i, k);

            for (char ch : substring) {
                hsum += alpha.find(ch);
            }

            int x = hsum % 26;
            ans += alpha[x];
        }

        return ans;
    }
};